deltacat/io/memcached_object_store.py

```
@@ -100,19 +108,19 @@ def _create_ref(self, uid, ip) -> str:
         return f"{uid}{self.SEPARATOR}{ip}"
 
     def _get_client_by_ip(self, ip_address: str):
-        if ip_address in self.client_cache:
-            return self.client_cache[ip_address]
-
-        base_client = Client((ip_address, self.port))
-        client = RetryingClient(
-            base_client,
-            attempts=3,
-            retry_delay=0.01,
-            retry_for=[MemcacheUnexpectedCloseError],
-        )
-
-        self.client_cache[ip_address] = client
-        return client
+        self.initialize_hasher()
+        storage_node_ip = self.hasher.get_node(ip_address)
+        if storage_node_ip:
+            base_client = Client((storage_node_ip, self.port))
+            client = RetryingClient(
+                base_client,
+                attempts=3,
+                retry_delay=0.01,
+                retry_for=[MemcacheUnexpectedCloseError, ConnectionResetError],
+            )
+            return client
+        else:
+            logger.warning(f"Failed to find storage_node_ip for {ip_address}")
```

can we raise an error instead?